jina/serve/runtimes/gateway/http/app.py

```diff
@@ -227,6 +228,88 @@ async def _render_custom_swagger_html(req: Request) -> HTMLResponse:
 
         app.add_route(docs_url, _render_custom_swagger_html, include_in_schema=False)
 
+    if not args.no_graphql_endpoint:
+        with ImportExtensions(required=True):
+            from dataclasses import asdict
+
+            import strawberry
+            from docarray import DocumentArray
+            from docarray.document.strawberry_type import (
+                Base64,
+                JSONScalar,
+                NdArray,
+                StrawberryDocument,
+            )
+            from strawberry.fastapi import GraphQLRouter
+
+            @strawberry.input
+            class StrawberryDocumentInput:
+                id: Optional[str] = None
+                parent_id: Optional[str] = None
+                granularity: Optional[int] = None
+                adjacency: Optional[int] = None
+                blob: Optional[Base64] = None
+                tensor: Optional[NdArray] = None
+                mime_type: Optional[str] = None
+                text: Optional[str] = None
+                weight: Optional[float] = None
+                uri: Optional[str] = None
+                tags: Optional[JSONScalar] = None
+                offset: Optional[float] = None
+                location: Optional[List[float]] = None
+                embedding: Optional[NdArray] = None
+                modality: Optional[str] = None
+
+            @strawberry.input
+            class JinaRequestModel:
+                """
+                Jina HTTP request model.
+                """
+
+                data: Optional[List[StrawberryDocumentInput]] = strawberry.field(
+                    default=None,
+                    description='Data to send, a list of dict/string/bytes that can be converted into a list of `Document` objects',
+                )
+                target_executor: Optional[str] = strawberry.field(
+                    default=None,
+                    description='A regex string represent the certain pods/deployments request targeted.',
+                )
+                parameters: Optional[JSONScalar] = strawberry.field(
+                    default=None,
+                    description='A dictionary of parameters to be sent to the executor.',
+                )
+                exec_endpoint: str = strawberry.field(
+                    default='/search',
+                    description='The endpoint string, by convention starts with `/`. '
+                    'All executors bind with `@requests(on="/foo")` will receive this request.',
```

Comment:
Maybe this is clearer, but maybe it is actually more confusing to the user. I'm not sure myself.
```suggestion
                exec_endpoint: str = strawberry.field(
                    default='/search',
                    description='The endpoint string, by convention starts with `/`. '
                    'All executors bind with `@requests(on=exec_endpoint)` will receive this request.',
```
```suggestion
                exec_endpoint: str = strawberry.field(
                    default='/search',
                    description='The endpoint string, by convention starts with `/`. '
                    'All executors bind with `@requests(on="/foo")` will receive this request.',
```